### PR TITLE
DISCO-1041 - Remove unused mediators

### PR DESCRIPTION
**Pull request by pepopowitz**
This addresses https://artsyproduct.atlassian.net/browse/DISCO-1041. 

While working on pagination changes for /collect, /collection/:id, /search, and /artist routes, we incurred some tech debt by leaving unused `mediator` instances around. This PR removes them. I added a _few_ types along the way, to make sure I was removing the right things.

---

Update: The PR this was depending (#2414) has been merged, so this one is all set!

### CODE REVIEW

File: src/Apps/Artist/Routes/Overview/Components/ArtworkFilter/ArtworkFilterArtworkGrid.tsx [hunk 1]
```diff
@@ -27,7 +27,7 @@ class Artworks extends Component<Props, LoadingAreaState> {
     isLoading: false,
   }
 
-  loadNext = (filters, mediator) => {
+  loadNext = (filters: FilterState) => {
```
Comment (damassi) [inline src/Apps/Artist/Routes/Overview/Components/ArtworkFilter/ArtworkFilterArtworkGrid.tsx]: 👌

File: src/Apps/Collect/FilterState.tsx [hunk 2]
```diff
@@ -138,17 +138,39 @@ export class FilterState extends Container<State> {
     return omitBy(this.state, isNil)
   }
 
-  setPage(page, _mediator) {
-    this.setState({ page })
+  previousQueryString = ""
+  pushHistory() {
```
Comment (damassi) [inline src/Apps/Collect/FilterState.tsx]: #minor, maybe a line break between these two?

**Comment by artsyit** [comment]
Deploy preview for *artsy-reaction* ready!

Built with commit f451c2c550adcb069954f542dbfd70b07ec9374a

https://deploy-preview-2415--artsy-reaction.netlify.com

**Comment by artsyit** [comment]
:rocket: PR was released in v16.11.20 :rocket:

**Pull request merged**